### PR TITLE
vendor: bump runc to v1.0.1

### LINKED ISSUE #102676: Pod stuck in ContainerCreating: Unit ...slice already exists
#### What happened:

Errors like this one 

> May 27 06:38:19.960408 ip-10-0-220-230 hyperkube[1448]: E0527 06:38:19.960361    1448 pod_workers.go:190] "Error syncing pod, skipping" err="failed to ensure that the pod: 5ac83c3f-0b16-4cf2-a3cb-f67c19cd0e16 cgroups exist and are correctly applied: failed to create container for [kubepods burstable pod5ac83c3f-0b16-4cf2-a3cb-f67c19cd0e16] : Unit kubepods-burstable-pod5ac83c3f_0b16_4cf2_a3cb_f67c19cd0e16.slice already exists." pod="openshift-machine-config-operator/machine-config-daemon-mm7gt" podUID=5ac83c3f-0b16-4cf2-a3cb-f67c19cd0e16

(when using cgroupDriver: systemd)

#### What you expected to happen:

No such errors

#### How to reproduce it (as minimally and precisely as possible):

I don't know for sure.

#### Anything else we need to know?:

This was introduced in k8s in #102147 and backported to 1.21 in #102196, so needs to be fixed in both master and `release-1.21`.

RH BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1965545

The cause is a regression in runc/libcontainer: https://github.com/opencontainers/runc/issues/2996

The fix is in https://github.com/opencontainers/runc/pull/2997, which should make its way into runc 1.0.0 GA.

Currently there is DNM PR to bump runc to the version with the fix: https://github.com/kubernetes/kubernetes/pull/102508, but we have decided (https://github.com/kubernetes/kubernetes/pull/102250#issuecomment-855922711) to wait until the release.

#### Environment:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by kolyshkin**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

Update runc/libcontainer to v1.0.1, fixing a few bugs, including the one that got 1.0.0 bump reverted
(see https://github.com/kubernetes/kubernetes/pull/103483).

For runc release notes, see
* 1.0.1: https://github.com/opencontainers/runc/releases/tag/v1.0.1
* 1.0.0: https://github.com/opencontainers/runc/releases/tag/v1.0.0

#### Which issue(s) this PR fixes:

Fixes: #102676

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

none

### CODE REVIEW

**Review by ehashman — COMMENTED** [review]
/ok-to-test
/triage accepted
/priority critical-urgent

/test pull-kubernetes-node-crio-cgrpv2-e2e
/test pull-kubernetes-node-kubelet-serial-crio-cgroupv1
/test pull-kubernetes-node-kubelet-serial-crio-cgroupv2
/test pull-kubernetes-node-kubelet-serial

**Comment by ehashman** [comment]
Note the serial tests are expected to fail but we wanna get signal: https://github.com/kubernetes/kubernetes/pull/103716#issuecomment-880683569

CRI-O e2e results (non-reporting) https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/103743/pull-kubernetes-node-crio-e2e/1416124153450205184

**Comment by BenTheElder** [comment]
also https://prow.k8s.io/pr-history/?org=kubernetes&repo=kubernetes&pr=103743 to see results over time for all jobs reporting or otherwise

FYI @hakman

File: staging/src/k8s.io/apiserver/go.mod [hunk 1]
```diff
@@ -22,6 +22,7 @@ require (
 	github.com/grpc-ecosystem/go-grpc-prometheus v1.2.0
 	github.com/munnerz/goautoneg v0.0.0-20191010083416-a7dc8b61c822
 	github.com/pquerna/cachecontrol v0.0.0-20171018203845-0dec1b30a021 // indirect
+	github.com/sirupsen/logrus v1.8.1 // indirect
```
Comment (BenTheElder) [inline staging/src/k8s.io/apiserver/go.mod:25]: suspect `hack/lint-dependencies.sh` will turn up something for this.

Comment (BenTheElder) [reply staging/src/k8s.io/apiserver/go.mod:25]: or not? https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/103743/pull-kubernetes-dependencies/1416124150098956288 seems happy.

**Comment by kolyshkin** [comment]
AFAIK serial tests are supposed to fail. @odinuge PTAL 🙏🏻

**Comment by endocrimes** [comment]
Went through failing tests and didn't come across any net-new failures (before the recent retest)

/lgtm

**Comment by dims** [comment]
the go.mod/go.sum changes look good to me. looks like one personal repo got switched over to a github org with the same license.

/approve
/lgtm

the CI jobs being 🔴 is concerning for sure. 

/assign @liggitt @smarterclayton for root OWNERS approval

**Comment by liggitt** [comment]
reviewing comparing to https://github.com/kubernetes/kubernetes/pull/102508

**Comment by liggitt** [comment]
/approve
dependency update looks good

/hold
for lgtm from node approver

**Comment by dims** [comment]
the CI failures look ok (agree with @endocrimes). We would need an ok from sig-node approver like @liggitt said.

thanks,
Dims

**Comment by dims** [comment]
/skip
/hold cancel
(@mrunalp is a node approver)

**Comment by dims** [comment]
@kolyshkin please file a cherry-pick to 1.22 release branch.

**Comment by kolyshkin** [comment]
> @kolyshkin please file a cherry-pick to 1.22 release branch.

Created: https://github.com/kubernetes/kubernetes/pull/103884

For 1.21, see https://github.com/kubernetes/kubernetes/pull/103746

**Comment by ehashman** [comment]
flake #103512

/test pull-kubernetes-integration

**Comment by hakman** [comment]
I see this change is being backported to all supported releases, so just wanted to mention that we noticed some instability in kOps periodic tests. When systemd cgroups were used with older versions of containerd (without runc 1.0.1) some random tests were failing. This was fixed by switching to a containerd version using runc 1.0.1, but would be nice to mention it in the release notes.

If you scroll back to 07-30, you should see how tests started failing once this was merged and then went back to green once kOps started using containerd 1.4.9:
https://testgrid.k8s.io/kops-misc#kops-aws-misc-amd64-conformance

**Comment by odinuge** [comment]
Hi,

Sorry that I haven't had time to review before now. Have had some time AFK after delivering my thesis.

> I see this change is being backported to all supported releases, so just wanted to mention that we noticed some instability in kOps periodic tests. When systemd cgroups were used with older versions of containerd (without runc 1.0.1) some random tests were failing. This was fixed by switching to a containerd version using runc 1.0.1, but would be nice to mention it in the release notes.
>
> If you scroll back to 07-30, you should see how tests started failing once this was merged and then went back to green once kOps started using containerd 1.4.9:
https://testgrid.k8s.io/kops-misc#kops-aws-misc-amd64-conformance

Yeah. K8s with systemd cgroup driver is still "broken" with this PR, since it does freeze the control groups, possibly causing strange behavior. And yeah, when doing that, runc pre v1.0.1 will permanently freeze containers, causing even more strange behavior. More in depth analysis here: https://github.com/kubernetes/kubernetes/issues/102676#issuecomment-880653430.

But thanks for reporting @hakman.

https://github.com/opencontainers/runc/pull/3143 should mitigate this problem, but we still need to get it fixed in runc and then get a release there, before merging it here... And yeah, that can potentially take some time.... :/

**Pull request merged**